autogpt/memory/milvus.py

```diff
@@ -13,28 +13,52 @@ def __init__(self, cfg) -> None:
         Args:
             cfg (Config): Auto-GPT global config.
         """
-        # connect to milvus server.
-        connections.connect(address=cfg.milvus_addr)
+        if cfg.milvus_uri is not None and cfg.milvus_uri != "":
+            connections.connect(
+                uri=cfg.milvus_uri,
+                user=cfg.milvus_username,
+                password=cfg.milvus_password,
+                secure=True,
+            )
+            self.index_params = {
+                "index_type": "AUTOINDEX",
+                "metric_type": "IP",
+                "params": {},
+            }
+        else:
+            # connect to milvus server.
+            connections.connect(
+                address=cfg.milvus_addr,
+                user=cfg.milvus_username,
```

Comment:
@chyezh Will this work if there is no username and password? (In case of [Milvus Lite](https://milvus.io/docs/milvus_lite.md))

Reply:
It works well in milvus standalone, but test do not cover the milvus lite. username and password is empty string by default in connect function. it seems that set milvus_username milvus_password to empty string by default but not None is a better choice. I change it in new commit.

Reply:
> @chyezh Will this work if there is no username and password? (In case of [Milvus Lite](https://milvus.io/docs/milvus_lite.md))

It should work well with Milvus lite.

Milvus and Milvus Lite can share the same client code.